include/socgraph.php

```diff
@@ -190,181 +187,166 @@ function poco_load_worker($cid, $uid, $zcid, $url) {
 	);
 
 }
+/**
+ * @brief Sanitize the given gcontact data
+ *
+ * @param array $gcontact array with gcontact data
+ *
+ * Generation:
+ *  0: No definition
+ *  1: Profiles on this server
+ *  2: Contacts of profiles on this server
+ *  3: Contacts of contacts of profiles on this server
+ *  4: ...
+ *
+ */
+function sanitized_gcontact(&$gcontact) {
```

Comment:
I disagree with the naming of the function and the usage of the reference operator.

I think the function should be named `sanitize_gcontact()` and return the sanitized gcontact data or `false` if a filter failed. Better yet, it should throw an Exception. Further down the road, it should be a method of the GContact class. In any case I don't think the reference operator is a good idea.